### ISSUE TITLE
Reranking Consume way too much memory space especially for MSMT17 dataset

### ISSUE THREAD

**Issue by FatimaZulfiqar** [KaiyangZhou/deep-person-reid#455]
Hi. I am trying to apply reranking on the MSMT17 dataset, which is one of the largest datasets for person reid. However, when I apply rerank it consumes way too much memory space and RAM gets exceeded due to many images in both query and gallery sets. I don't have extremely powerful resources with plenty of RAM space. The highest space I have right now is 25GB. Is there a way to make adjustments in the rerank.py file so that memory does not get exceeded to that much extent? Need help and guidance in this regard. 

Basically, I need an algorithm for reranking that is space efficient.
